### DISCUSSION TITLE
Remove the gray outlines around active tree elements

### DISCUSSION THREAD

**Discussion by passivestar** [godotengine/godot-proposals#9123] [Editor]
The gray outline around the tree active element is called a "cursor"

Since https://github.com/godotengine/godot/pull/86218 was merged in december it's no longer possible to see that cursor separately from the selected item (to my knowledge), because the arrow keys now change the actual selection and not just the cursor

<img width="448" alt="image" src="https://github.com/godotengine/godot-proposals/assets/60579014/b8842a23-14fd-4899-b651-8ef916ede6f0"><br>

Now that that's the case, the gray outline is no longer necessary and can be removed for less visual noise and for consistency with the look of active elements in the other parts of the editor's UI, as a lot of them don't have that gray outline:

<img width="443" alt="image" src="https://github.com/godotengine/godot-proposals/assets/60579014/2106a75b-7732-4bb2-a945-9e880fc56ea3"><br>

<img width="353" alt="image" src="https://github.com/godotengine/godot-proposals/assets/60579014/29a8b1bd-8dc0-4e66-9abe-5307a8408ed0"><br>

<img width="323" alt="image" src="https://github.com/godotengine/godot-proposals/assets/60579014/096857e2-a842-41ce-a156-7299b0b0258c"><br>

In the short-term this can be done by just swapping out the style for the cursor to an empty stylebox

In the long-term perhaps the concept of the cursor itself could be revisited. I don't know the internals of Godot well enough to go as far as saying that the cursor needs to be removed altogether from the codebase, but at least as a user of the engine I can say that it feels like it doesn't really contribute anything to the UX

**Comment by Calinou** [comment]
I agree with removing this outline, and also making the select highlight cover the entire line. Right now, it doesn't span the entire line in the Scene tree dock but it does in the signals dock you shown.

**Comment by alfredbaudisch** [comment]
Just quoting what [I said on Twitter](https://twitter.com/AlfredBaudisch/status/1759281473272664123) to @passivestar: keep all the outlines toggeable with "Draw Extra Borders", it's very important for visual accessibility and readability, just like most popular IDEs' "high contrast" themes.

And to keep things consistent, **all outlines should come disabled by default and appear only if "Draw Extra Borders" is on**?

![GGo4wDbW8AA2LTe](https://github.com/godotengine/godot-proposals/assets/248383/88b1c23c-27bd-45d0-a61d-4eac16aebf2c)

In any case, even "Draw Extra Borders" is currently inconsistent - where are the outlines here, for example?

![image](https://github.com/godotengine/godot-proposals/assets/248383/fabac42a-3230-4981-9bc8-957ef6ab658a)

**Reply by passivestar** [reply]
Forwarding my reply from twitter then:

> yeah I think that would be a sensible thing to do. although I must say that *adding* outlines to elements that didn't have them before is a lot more work than removing the tree cursor, so those things might not happen simultaneously

**Comment by davthedev** [comment]
The cursor box is still there because the tree is in multiple select mode. There is no box on Tree controls that are configured in Single or Row select modes.
But, unlike the ItemList, as you mention, the arrows do not move the cursor separately. Something that should be fixed because the ability to select multiple nodes in the scene tree is important IMHO. I can edit the same property using multiple objects at the same time.
I think the ability to move the cursor separately on the Tree controls should be restored, if the said Tree is in multiple select mode, making it consistent with the ItemList.

Any Tree can already have its whole row be highlighted - it is sufficient to put it in Row selection mode. But this is not compatible with multiple selection mode yet - something that can be developed!

**Comment by passivestar** [comment]
Superseded by https://github.com/godotengine/godot/pull/111118
